### PR TITLE
perf(dsv4 hc_head): pure-AIC split-K, fused pre, tensor-level reduce

### PR DESCRIPTION
**Pull request by Hzfengsy**
## Summary
- Dedicated `hc_head_cast` scope streams x (BF16) -> x_fp32 (FP32) once; both the head-projection matmul and the inv_rms reduce read FP32 directly, so the matmul is a clean pure-AIC (cube-only) kernel instead of mixed cube+cast.
- Split-K head projection (`LINEAR_OK=8`): zero-seed `mixes_raw`, then atomic-add per-(token-tile, K-slice) FP32 partials to fill idle cube cores and shorten each core's `matmul_acc` chain.
- Fuse scale + sigmoid + transpose into one `hc_head_pre_fused` scope (replaces the separate `hc_head_pre` + `hc_head_transpose_pre`); the mix/pre intermediates never round-trip through GM.
- Tensor-level `hc_head_reduce` (replaces the low-level `pl.load`/`pl.store` form).
- `T_TILE` 16 -> 8; minor: `--enable-l2-swimlane` becomes an int mode (0/1/2).
- Golden-validated (`ratio_allclose`). ~3x faster by AICore trace-wall on the standalone hc_head decode test (~199us -> ~66us, best-of-5, device 0).

## Related Issues
None.

### CODE REVIEW

File: models/deepseek/v4/hc_head.py [hunk 1]
```diff
@@ -25,13 +25,37 @@
...
+# median ~128-134us.
+LINEAR_OK = 8
 RMS_K_BLOCKS = HC_DIM // RMS_K_CHUNK
 LINEAR_K_BLOCKS = HC_DIM // LINEAR_K_CHUNK
 D_BLOCKS = D // D_CHUNK
+NUM_T_TILES = T // T_TILE
+NUM_LINEAR_T_TILES = T // LINEAR_T_TILE
+LINEAR_K_PER_SPLIT = HC_DIM // LINEAR_OK
+LINEAR_CHUNKS_PER_SPLIT = LINEAR_K_PER_SPLIT // LINEAR_K_CHUNK
+assert HC_DIM % LINEAR_OK == 0 and LINEAR_K_PER_SPLIT % LINEAR_K_CHUNK == 0
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply models/deepseek/v4/hc_head.py:58]: Fixed in 0a35868 — added `assert HC_MULT == 4` and `assert T % T_TILE == 0 and T % LINEAR_T_TILE == 0 and T % TRANSPOSE_T_TILE == 0` after the existing K-divisibility assert.

File: models/deepseek/v4/hc_head.py [hunk 2]
```diff
@@ -44,111 +68,101 @@ def hc_head(
...
+    # NUM_T_TILES * LINEAR_OK tasks -- each owns one token-tile and one 1/OK
+    # K-slice, reduces it, and atomic-adds its [T_TILE, HC_PAD] FP32 partial.
+    # Token tiles touch disjoint rows, so only the LINEAR_OK tasks per row block
+    # contend; the seed write -> atomic RMW WAW dependency orders seed first.
+    with pl.at(level=pl.Level.CORE_GROUP, name_hint="hc_head_seed"):
+        for tc in pl.range(NUM_T_TILES):
+            ts0 = tc * T_TILE
+            mixes_raw[ts0 : ts0 + T_TILE, 0:HC_PAD] = pl.full(
+                [T_TILE, HC_PAD], dtype=pl.FP32, value=0.0
+            )
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply models/deepseek/v4/hc_head.py]: Fixed in 0a35868 — hc_head_seed now dispatches via pl.spmd(NUM_T_TILES) so the zero-fill parallelizes across cores instead of a single-core loop.

File: models/deepseek/v4/hc_head.py [hunk 3]
```diff
@@ -44,111 +68,101 @@ def hc_head(
...
-                target_memory=pl.MemorySpace.Vec,
+        for t0 in pl.pipeline(0, T, TRANSPOSE_T_TILE, stage=2):
+            scaled = pl.row_expand_mul(
+                mixes_raw[t0 : t0 + TRANSPOSE_T_TILE, 0:HC_PAD],
+                inv_rms[t0 : t0 + TRANSPOSE_T_TILE, 0:1],
+            )
+            logits = pl.add(
+                pl.mul(scaled, scale),
+                pl.col_expand_mul(pl.full([TRANSPOSE_T_TILE, HC_PAD], dtype=pl.FP32, value=1.0), base),
             )
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply models/deepseek/v4/hc_head.py:149]: Fixed in 0a35868 — replaced the col_expand_mul + full-1.0 tensor with pl.col_expand(scaled, base) (fewer UB allocs / vector instrs).

File: models/deepseek/v4/hc_head.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply models/deepseek/v4/hc_head.py:58]: Fixed in 0a35868 — added assert T % T_TILE == 0 and T % LINEAR_T_TILE == 0 and T % TRANSPOSE_T_TILE == 0 next to the K-divisibility assert.

File: models/deepseek/v4/hc_head.py [hunk 4]
```diff
@@ -244,14 +258,19 @@ def init_hc_head_fn():
                         choices=["a2a3", "a2a3sim", "a5", "a5sim"])
     parser.add_argument("-d", "--device", type=int, default=0)
     parser.add_argument("--seed", type=int, default=0)
-    parser.add_argument("--enable-l2-swimlane", action="store_true", default=False)
+    # Int mode (0=off; 1=timing only, most accurate; 2=timing + dep graph, two runs).
+    # `nargs="?"` so a bare `--enable-l2-swimlane` -> mode 1 (int, not bool True).
+    parser.add_argument("--enable-l2-swimlane", type=int, nargs="?", const=1, default=0)
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply models/deepseek/v4/hc_head.py]: Fixed in 0a35868 — added choices=(0, 1, 2) to the --enable-l2-swimlane argument.
